tools/ds-identify

```diff
@@ -1586,6 +1586,28 @@ dscheck_VMware() {
     return "${DS_NOT_FOUND}"
 }
 
+WSL_cloudinit_dir_in() {
+    _RET=""
+    for m in "$@"; do
+        cmdexe="$m/Windows/System23/cmd.exe"
+        if command -v "$cmdexe" > /dev/null 2>&1; then
+            cloudinitdir=$(/init "$cmdexe" /c echo %USERPROFILE% 2>/dev/null)
+            if [ -n "$cloudinitdir" ]; then
+                val=$(wslpath -au "$cloudinitdir") && _RET="$val"
+                return $?
+            fi
+        fi
+    done
+
+    return 1
+}
+
+WSL_instance_name() {
+    instance_name=$(wslpath -am /)
+    val=$(basename "$instance_name")
```

Comment:
We should be able to avoid this external dependency and shelling out by making use of the following builtin:
```

$ echo "$fname"
/usr/bin/sort
$ echo "$(basename $fname)"
sort
$ echo "${fname##*/}"
sort
```